projects/storefrontlib/cms-components/misc/icon/icon.component.ts

```diff
@@ -41,87 +42,97 @@ export class IconComponent {
    * The cxIcon directive is bound to the icon type. You can feed the `ICON_TYPE` to
    * accomplish a configurable button in the UI.
    */
-  @Input() set cxIcon(type: ICON_TYPE) {
+  @Input() set cxIcon(type: ICON_TYPE_STRING) {
     this.setIcon(type);
   }
 
   /**
    * The type input parameter is bound to the icon type. You can feed the `ICON_TYPE` to
    * accomplish a configurable button in the UI.
    */
-  @Input() set type(type: ICON_TYPE) {
+  @Input() set type(type: ICON_TYPE_STRING) {
+    // XXX this clashes with the type attribute of some html form elements, e.g. input or button
```

Comment:
I guess it'd be preferable to prefix all html stuff with `cx`. Maybe deprecate this and introduce a `cxType` setter instead? (Has little to do with this PR though.)

Reply:
If ever we rename things it will have to wait to a major version release because it's a breaking change.